x-pack/plugins/security_solution/public/common/containers/sourcerer/index.tsx

```diff
@@ -25,12 +27,31 @@ export const useInitSourcerer = (
   const initialTimelineSourcerer = useRef(true);
   const initialDetectionSourcerer = useRef(true);
   const { loading: loadingSignalIndex, isSignalIndexExists, signalIndexName } = useUserInfo();
-  const getConfigIndexPatternsSelector = useMemo(
-    () => sourcererSelectors.configIndexPatternsSelector(),
+  const getDefaultIndexPatternSelector = useMemo(
+    () => sourcererSelectors.defaultIndexPatternSelector(),
     []
   );
-  const ConfigIndexPatterns = useDeepEqualSelector(getConfigIndexPatternsSelector);
+  const defaultIndexPattern = useDeepEqualSelector(getDefaultIndexPatternSelector);
+  const { addError } = useAppToasts();
 
+  useEffect(() => {
+    const { id, ...rest } = defaultIndexPattern;
+    if (id === null) {
+      // if id is null, rest is error
+      addError(rest, {
+        title: i18n.translate('xpack.securitySolution.sourcerer.permissions.title', {
+          defaultMessage: 'Write role required to generate data',
+        }),
+        toastMessage: i18n.translate('xpack.securitySolution.sourcerer.permissions.toastMessage', {
+          defaultMessage:
+            'Users with write permission need to access the Elastic Security app to initialize the app source data.',
```

Comment:
Maybe change `access` to `visit`?

Reply:
I had the tech writers come up with this one for me so gonna keep it, thanks!